SYCL/BFloat16/bfloat16_type.cpp

```diff
@@ -1,8 +1,8 @@
 // UNSUPPORTED: hip
 // RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
-// Currently the feature is supported only on CPU and GPU, natively or by
-// software emulation.
-// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// Currently the feature is supported only on GPU, natively or by software
```

Comment:
It would be nice to say something about AOT here since the limitation is specific to AOT.  Don't you also have a problem on GPU when the GPU does not have native support?

Reply:
This is a test, and it is not doing AOT, so why bring that up?

Reply:
When we have compiler driver support for AOT everywhere, this test can be extended to use both JIT and AOT.

Reply:
Oh, I was confused about the test.  I thought it was doing AOT.  If it is a JIT mode test, why doesn't CPU work?

Reply:
Need to look into it. CPU should use the emulation method always. I thought the test failed because not all CPUs support bfloat16.

Reply:
OK, so we'll fix the problem and re-enable the test before merging?

Reply:
Building for the _CPU device_ is essentially an AOT build. Currently, AOT builds always pull in the native bfloat16 library, which isn't supported on the CPU.
We will have to defer _CPU device_ support for bfloat16 until the driver is changed to fully support AOT. The driver will select native bfloat16 library only for Xe GPUs and fallback bfloat16 library for other GPUs and the CPU.

Reply:
Something doesn't sound right.  You can definitely run on the CPU device without AOT compiling.  For example:

```
$ clang++ -fsycl -o very-simple very-simple.cpp    # Compiles without AOT
$ SYCL_DEVICE_FILTER=level_zero:gpu ./very-simple  # Runs on GPU
$ SYCL_DEVICE_FILTER=opencl:cpu ./very-simple      # Runs on CPU
```

Is there something weird about `%CPU_RUN_PLACEHOLDER` that forces an AOT compile?

Reply:
_CPU device_ support is not completely broken, but just for bfloat16.
This is not a regression. Even earlier, CPU device did not work with bfloat16.

When doing JIT compilation for CPU there is no target device check. The same libraries are always linked in. Currently for bfloat16 this is the native library (so that AOT for PVC works). When AOT is fixed to check which device is being compiled for, and the appropriate bfloat16 library is linked in, that will also correct the JIT case for CPU.

Reply:
I have made some temporary changes in the compiler driver to support AOT.

Reply:
Cool!  Does that mean we can re-enable testing on CPU?  BTW, shouldn't there also be a `GPU_RUN_PLACEHOLDER` here too?

Reply:
Yes, I've now enabled testing on CPU and GPU.

Reply:
Nice!